### PR TITLE
add Default GameRules

### PR DESCRIPTION
**Pull request by WaitingIdly**
changes in this PR:
- add the ability to set the default GameRules. can be done one-by-one or en-mass. 
- when a gamerule that doesnt exist tries to be added, log a warning unless the user has run `gameRule.setWarnNewGameRule(false)`.
- add a command `/gs applyDefaultGameRules` to apply these in-game.
- add examples for this
- the default gamerules are applied when `WorldEvent.CreateSpawnPosition` is fired - as far as i can tell, it is only fired once, when the world is created. other places that do similar things listen to `FMLServerStartingEvent` (applies on every load), `WorldEvent.Load` (applies on every load), and a particularly daring one did `EntityJoinWorldEvent`. 
	- i tested `WorldEvent.CreateSpawnPosition` on both client and server, and it only runs if the `level.dat` does not exist (something which tripped me up on the server for a bit due to always terminating the run instead of stopping the server).
- fix the server using the `run/examples` folder instead of `examples` when `groovyscript.use_examples_folder` is set to true. bug only affects us.

### CODE REVIEW

File: src/main/java/com/cleanroommc/groovyscript/compat/vanilla/GameRule.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import com.cleanroommc.groovyscript.registry.NamedRegistry;
+import net.minecraft.world.GameRules;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class GameRule extends NamedRegistry implements IScriptReloadable {
+
+    private static final String LOG_MESSAGE = "Could not find an already existing rule with the name {}. This may be intentional! If it is, you can disable this via `gameRule.setWarnNewGameRule(false)`";
+    private final Map<String, String> defaultGameRules = new HashMap<>();
```
Comment (brachy84) [inline src/main/java/com/cleanroommc/groovyscript/compat/vanilla/GameRule.java:16]: fastutil map?

**Review by brachy84 — CHANGES_REQUESTED** [review]
epic

File: src/main/java/com/cleanroommc/groovyscript/compat/vanilla/GameRule.java (hunk 1 above)
Comment (WaitingIdly) [reply src/main/java/com/cleanroommc/groovyscript/compat/vanilla/GameRule.java:16]: uh, its object to object, i was under the understanding that either `Object2ObjectOpenHashMap` or `HashMap` are fine, and theres little difference. is this incorrect?

Comment (brachy84) [reply src/main/java/com/cleanroommc/groovyscript/compat/vanilla/GameRule.java:16]: fastutil is always the better choice for hash maps

**Review by brachy84 — APPROVED** [review]

**Pull request merged**